New Integration: Cilium (#4665)
DataDog/integrations-core: cilium/README.md

# Agent Check: Cilium

## Overview

This check monitors [Cilium][1] through the Datadog Agent.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][2] for guidance on applying these instructions.

### Installation

The Cilium check is included in the [Datadog Agent][2] package.
No additional installation is needed on your server.

### Configuration

1. Edit the `cilium.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your cilium performance data. See the [sample cilium.d/conf.yaml][3] for all available configuration options.

2. [Restart the Agent][4].

### Validation

[Run the Agent's status subcommand][5] and look for `cilium` under the Checks section.

## Data Collected

### Metrics

> Cilium does not include any metrics.

Reviewer: ```suggestion
See [metadata.csv][x] for a list of metrics provided by this check.
```
And then add link.
We should update this on the new integration template

### Service Checks

Cilium does not include any service checks.

### Events

Cilium does not include any events.

## Troubleshooting

Need help? Contact [Datadog support][6].

> [1]: **LINK_TO_INTEGRATION_SITE**

Reviewer: Fix

# Agent Check: Cilium

## Overview

This check monitors [Cilium][1] through the Datadog Agent. The integration can either collect metrics from the `cilium-agent` or `cilium-operator`.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][2] for guidance on applying these instructions.

### Installation

> The Cilium check is included in the [Datadog Agent][2] package but requires additional setup steps to export Prometheus metrics.

Reviewer 2: ```suggestion
The Cilium check is included in the [Datadog Agent][2] package, but it requires additional setup steps to export Prometheus metrics.
```

> 1. In order to collect `cilium-agent` metrics, the Prometheus metrics must be enabled by either:

Reviewer 2: ```suggestion
1. In order to collect `cilium-agent` metrics, you must enable Prometheus metrics. There are two ways to do this. Either:
```

>     * Deploy Cilium with the `global.prometheus.enabled=true` Helm value set.

Reviewer 2: ```suggestion
    * Deploy Cilium with the `global.prometheus.enabled=true` Helm value set, or:
```

    * Add `--prometheus-serve-addr=:9090` to the `args` section of the Cilium DaemonSet config.
        ```
        [...]
            spec:
                containers:
                - args:
                    - --prometheus-serve-addr=:9090

        ```

> 2. In order to collect `cilium-operator` metrics, the Prometheus metrics must be enabled by adding `- --enable-metrics` to the args section of the Cilium DaemonSet config.

Reviewer 2: just to clarify -- you can't enable this in Helm?

Author: Hi! Yes you can, I can also update with those instructions. The following were just the methods I used.

### Configuration

> 1. Edit the `cilium.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your cilium performance data. See the [sample cilium.d/conf.yaml][3] for all available configuration options.

Reviewer 2: ```suggestion
1. Edit the `cilium.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your Cilium performance data. See the [sample cilium.d/conf.yaml][3] for all available configuration options.
```

# Agent Check: Cilium

## Overview

This check monitors [Cilium][1] through the Datadog Agent. The integration can either collect metrics from the `cilium-agent` or `cilium-operator`.

## Setup

Follow the instructions below to install and configure this check for an Agent running on a host. For containerized environments, see the [Autodiscovery Integration Templates][2] for guidance on applying these instructions.

### Installation

The Cilium check is included in the [Datadog Agent][3] package, but it requires additional setup steps to expose Prometheus metrics.

1. In order to enable Prometheus metrics in both the `cilium-agent` and `cilium-operator`, deploy Cilium with the `global.prometheus.enabled=true` Helm value set, or:

2. Separately enable Prometheus metrics in the `cilium-agent`:
    * Add `--prometheus-serve-addr=:9090` to the `args` section of the Cilium DaemonSet config.
        ```
        [...]
            spec:
                containers:
                - args:
                    - --prometheus-serve-addr=:9090
        ```
    or in the `cilium-operator`:

>     * Add `- --enable-metrics` to the `args` section of the Cilium deployment config.

Reviewer 3: ```suggestion
    * Add `--enable-metrics` to the `args` section of the Cilium deployment config.
```
